datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -168,6 +164,22 @@ protected AbstractKafkaBasedConnectorTask(KafkaBasedConnectorConfig config, Data
     _startOffsets = Optional.ofNullable(_datastream.getMetadata().get(DatastreamMetadataConstants.START_POSITION))
         .map(json -> JsonUtils.fromJson(json, new TypeReference<Map<Integer, Long>>() {
         }));
+
+    // if this datastream want to use specific start offsets, we need to have the auto.offset.reset config
+    // set to "none" in the kafka consumer configs, so that the Kafka consumer throws NoOffsetForPartiionException
+    // upon first poll, to be handled by seeking to teh startOffsets
+    if (_startOffsets.isPresent()) {
+      _logger.info("Datastream contains startOffsets, override {} with value {} (was {} in the provided configs)",
```

Comment:
nit: I was originally conflicted about the change because both start position and reset strategy are overridable values and we are basically giving higher precedence to start position by adjusting the reset strategy. But on more thought I do feel like this is the right thing to do, given that reset strategy isn't something a client will provide when they have already specified the start position. Although, it would be confusing is if they provided both as it wouldn't be possible to figure out which one they really wanted to set, I do feel like start position somehow feels like a more granular thing to specify and so should probably take precedence. 
Can we however fetch offset reset strategy first so that in the logs it can be made clear that that was ignored because startPosition was also set?

Reply:
+1
We could perhaps add some kind of validation that start positions cannot be present if the auto.offset.reset strategy is set to anything other than "none" if we feel the need to be careful, but I'll leave the decision to add this or not to you.

Reply:
I was thinking about validation, but then the stream task would just die, right? I wasn't terribly sure if this was a good idea since clients are not necessarily good at figuring out what's happening on the server, but may be in such cases they are, and may be SREs should in fact get alerted for bad configs.

Reply:
@DEEPTHIKORAT what about checking this during a) initialize datastream and b) datastream update? This way both stream creation/update will fail. 
Again don't have a strong opinion on whether we should add this or not, but it's just a thought if we're worried about explaining the behavior when both metadata fields are present.

Reply:
I agree with Deepthi that failing might not be the right thing to do. And that was exactly my thinking @DEEPTHIKORAT  that when users set the start offsets, they mean to use it. I am also doing exactly as you suggest: I am logging the original value before the update. Here is a sample log from the test:

`2020-12-17 16:27:41 INFO  KafkaConnectorTask:176 - Datastream contains startOffsets, override auto.offset.reset with value none (was earliest in the provided configs)`

I will attempt to reword to make it clearer

Reply:
I think that is a nice idea @somandal. @shenodaguirguis feel free to check in this version since Sonam has approved it as well, but we can consider making the change Sonam suggested in the future perhaps. At least then it is a user-facing failure and they won't be caught unaware.

Reply:
we can have a separate change for this